src/Data/Rational/Tactic/RingSolver.agda

```diff
@@ -0,0 +1,46 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Automatic solvers for equations over rationals
+------------------------------------------------------------------------
+
+-- See README.Integer for examples of how to use this solver
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Rational.Tactic.RingSolver where
+
+open import Agda.Builtin.Reflection using (Term; TC)
+
+open import Agda.Builtin.Int using (Int; negsuc; pos)
+open import Data.Nat.Base using (zero; suc)
+open import Data.Maybe.Base using (Maybe; just; nothing)
+open import Data.Rational.Base using (ℚ; 0ℚ; mkℚ)
+open import Data.Rational.Properties using (+-*-commutativeRing)
+open import Level using (0ℓ)
+open import Data.Unit using (⊤)
+open import Relation.Binary.PropositionalEquality using (_≡_; refl)
+
+import Tactic.RingSolver as Solver using (solve-macro; solve-∀-macro)
+import Tactic.RingSolver.Core.AlmostCommutativeRing as ACR
+
+------------------------------------------------------------------------
+-- A module for automatically solving propositional equivalences
+-- containing _+_ and _*_
+
+ring : ACR.AlmostCommutativeRing 0ℓ 0ℓ
+ring = ACR.fromCommutativeRing +-*-commutativeRing f
+  where
+   f : (x : ℚ) → Maybe (0ℚ ≡ x)
+   f (mkℚ (pos 0)       0       _) = just refl
+   f (mkℚ (pos 0)       (suc _) _) = nothing
+   f (mkℚ (pos (suc _)) _       _) = nothing
+   f (mkℚ (negsuc _)    _       _) = nothing
```

Comment:
I'd be strongly tempted to lift this out as a definition in `Data.Rational.Base`, or else as a property (cf. `Relation.Binary.WeaklyDecidable`, for which we currently lack analogues at arity 0, 1) in `Data.Rational.Properties`.

Suggested name for such a refactored `f`: `isZero-weakly-decidable`?

That way, the `import`s for your solver modules then become very much simplified, in favour of those that are already currently used by the `Rational.*` modules...

... and there's (potentially) a downstream benefit in being able to reuse the definition of the corresponding `Unnormalised` function in the definition of this one.